docs/reference/snapshot-restore/register-repository.asciidoc

```diff
@@ -294,3 +294,27 @@ subsequently cleaned up.
 Please note that most of the cleanup operations executed by this endpoint are automatically executed when deleting any snapshot from a
 repository. If you regularly delete snapshots, you will in most cases not get any or only minor space savings from using this functionality
 and should lower your frequency of invoking it accordingly.
+
+[discrete]
+[[snapshots-repository-backup]]
+=== Repository backup
```

Comment:
I'm in two minds about this section. We don't have any tests to support these claims today AFAIK, and it seems like they might be pretty tricky to write. How comfortable are we with writing this in the docs without such tests?

Reply:
I have an idea for a way to write tests for this but have encountered some unrelated issues along the way.

Reply:
I'm ok with this section. I'm not sure what you would like to test here. Isn't it just about having a repo mounted, then unmount it, and then remount it and see if everything works?

Reply:
No I'd like to take a copy of the repo either (a) atomically, while snapshots/deletes/... are ongoing or (b) while it's readonly and verify that the copy is usable as a repository too. You can only unregister a repository while there's no ongoing activity which is a much weaker claim.